jetty-core/jetty-osgi/src/main/java/org/eclipse/jetty/osgi/PackageAdminServiceTracker.java

```diff
@@ -36,31 +37,44 @@
  * When the PackageAdmin service is activated we can look for the fragments
  * attached to this bundle and do a fake "activate" on them.
  * <p>
- * See particularly the jetty-ee9-osgi-boot-jsp fragment bundle that uses this
- * facility.
+ * See particularly the jetty-ee9-osgi-boot-jsp fragment bundle that uses this facility.
  */
+@SuppressWarnings("deprecation")
 public class PackageAdminServiceTracker implements ServiceListener
 {
-    private static Logger LOG = LoggerFactory.getLogger(PackageAdminServiceTracker.class);
-    private BundleContext _context;
+    private static final Logger LOG = LoggerFactory.getLogger(PackageAdminServiceTracker.class);
 
-    private List<BundleActivator> _activatedFragments = new ArrayList<>();
+    private final String _environment;
+    private final BundleContext _context;
+    private final List<BundleActivator> _activatedFragments = new ArrayList<>();
 
-    private boolean _fragmentsWereActivated = false;
-
-    // Use the deprecated StartLevel to stay compatible with older versions of
-    // OSGi.
+    // Use the deprecated StartLevel to stay compatible with older versions of OSGi.
     private StartLevel _startLevel;
-
     private int _maxStartLevel = 6;
 
-    public static PackageAdminServiceTracker INSTANCE = null;
+    private static final Map<String, PackageAdminServiceTracker> INSTANCES = new ConcurrentHashMap<>();
```

Comment:
I don't think this is the solution we need. Each `Bundle` that contains an `EEXActivator` class should have it's own `PackageAdminServiceTracker` so that it can activate it's own associated fragment bundles. This solution would keep references to the `PackageAdminServiceTrackers` from all of the other bundles that contain an `EEXActivator`.  This needs some more refactoring, let me think on it a bit and get back to you.

Reply:
Feel free to take over this one.
I've been struggling to understand why ee9 fails, but ee10 doesn't on this branch anyway.